frontend/src/components/core/Block/ElementNodeRenderer.tsx

```diff
@@ -380,6 +371,23 @@ const RawElementNodeRenderer = (
       return <Video width={width} element={node.element.video as VideoProto} />
 
     // Widgets
+    case "arrowDataFrame": {
+      const arrowProto = node.element.arrowDataFrame as ArrowProto
+      widgetProps.disabled = widgetProps.disabled || arrowProto.disabled
+
+      return (
+        <ArrowDataFrame
+          element={arrowProto}
+          data={node.quiverElement as Quiver}
+          width={width}
+          height={height}
+          {...(arrowProto.id && {
+            key: arrowProto.id,
```

Comment:
What's the id in this case? Is this like a hash of the data or something? Will this be for both the element and the widget form? A comment might be nice here

Reply:
It's the widget key (as with other widgets). The arrow dataframe can be used as a widget (`st.data_editor`) or as an element (`st.dataframe`). In case of `st.dataframe`, the id will be undefined... so we don't set it. I added a comment explaining this.